rust/src/test.rs

```diff
@@ -18,16 +18,19 @@ pub enum TestError {
 }
 
 /// A single minimal test case.
-pub struct TestCase {
+pub struct TestCase<const SERIALIZED: bool = false> {
     pub name: &'static str,
     pub description: &'static str,
     pub require_root: bool,
-    pub fun: fn(&mut TestContext),
```

Comment:
If you change this to be
```rust
enum TestFunc {
    Serialized(fn(&mut SerializedTestContext)),
    NonSerialized(fn(&mut TestContext))
}
pub struct TestCase {
    ...
   pub fun: TestFunc
}
```
then you'll only have a single `TestCase` type, which means only a single inventory to collect.  It would simply the code in main.rs.
pub fn: